include/pybind11/pybind11.h

```diff
@@ -422,6 +422,7 @@ class cpp_function : public function {
             }
         } catch (const error_already_set &)      {                                                 return nullptr;
         } catch (const index_error &e)           { PyErr_SetString(PyExc_IndexError,    e.what()); return nullptr;
+        } catch (const value_error &e)           { PyErr_SetString(PyExc_ValueError,    e.what()); return nullptr;
```

Comment:
Could you also add a note about this new kind of exception to ``docs/advanced.rst``?